lib/core/src/sdk.rs

```diff
@@ -1024,8 +1058,12 @@ impl LiquidSdk {
                     created_at: utils::now(),
                     state: PaymentState::Created,
                     refund_private_key: keypair.display_secret().to_string(),
+                    is_local: true,
                 };
                 self.persister.insert_send_swap(&swap)?;
+                self.sync_service
```

Comment:
This is not atomic so we can end up with some data that is not synced upstream.
Perhaps we can have an updated_timestamp and synced_timestamp and always query for these records that have their updated_timestamp larger than the synced_timestamp as candidates to sync?

Also where is the server record id is saved in the client to associate a swap with a record id?

Reply:
There's a new method that needs to be added in the interface. As you said in the comment above, `get_changes_since` is an internal method. I will add a `get_latest_changes`, which uses the client's `latestRecordId`.

For the first question, that is what `get_latest_changes` will be for. I haven't added it to the SDK init yet, though the idea is that on startup the sync service will send its `latestRecordId` (stored in a new `settings` table on the DB, it can be renamed afterwards), and try and pull all the upstream changes. 

This mechanism + listening for new updates at startup ensures that, when we later call `set_record`, we are always synced with upstream and can actually push changes. If we are not, set_record returns error (and we probably want to trigger `get_latest_changes` again).